### ISSUE TITLE
JSON-RPC

### ISSUE THREAD

**Issue by odrotleff** [clemahieu/raiblocks#589]
Is there a reason the wallet RPC protocol does not follow the JSON-RPC standard? Implementation would be much easier if one could use JSON-RPC libraries.

**Comment by cryptocode** [comment]
It's due to Boost. Some discussion here: https://github.com/clemahieu/raiblocks/pull/526

**Comment by odrotleff** [comment]
That is also disappointing, but I meant following the JSON-RPC specification (http://www.jsonrpc.org/specification).

> A rpc call is represented by sending a Request object to a Server. The Request object has the following members:
>
>* jsonrpc
>A String specifying the version of the JSON-RPC protocol. MUST be exactly "2.0".
> * method
>A String containing the name of the method to be invoked. Method names that begin with the word rpc followed by a period character (U+002E or ASCII 46) are reserved for rpc-internal methods and extensions and MUST NOT be used for anything else.
>* params
>A Structured value that holds the parameter values to be used during the invocation of the method. This member MAY be omitted.
>* id
>An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [1] and Numbers SHOULD NOT contain fractional parts [2]

**Comment by cryptocode** [comment]
Yeah, sure, supporting json-rpc would be nice, but the first step would be to introduce a proper json library. And all this breaks existing clients, so it's definitely a new RPC protocol version.

**Comment by odrotleff** [comment]
Gotcha, agreed. Is someone already working on this? My C++ is a little rusty :)

**Comment by cryptocode** [comment]
@odrotleff Not that I'm aware of. One idea that's been talked about is to move the RPC stuff out of core and reimplement it in terms of a C api  (I'm working on one api proposal, others are possibly too)

**Comment by rkeene** [comment]
There has been work on this as part of #947

**Comment by zhyatt** [comment]
Related to #1434
